client/lcd/root.go

```diff
@@ -62,21 +66,28 @@ func ServeCommand(cdc *codec.Codec) *cobra.Command {
 			})
 
 			var cleanupFunc func()
+
+			// TODO: re-enable insecure mode once #2715 has been addressed
 			if viper.GetBool(flagInsecure) {
-				listener, err = tmserver.StartHTTPServer(
-					listenAddr, handler, logger,
-					tmserver.Config{MaxOpenConnections: maxOpen},
+				fmt.Println(
+					"Insecure mode is temporarily disabled, please locally generate an " +
+						"SSL certificate to test. Support will be re-enabled soon!",
 				)
-				if err != nil {
-					return
-				}
+				// listener, err = tmserver.StartHTTPServer(
```

Comment:
Why do we need to comment this? Just return an error (don't fail silently)

Reply:
Return an error where?

Reply:
If I return an error after, I'll have to restructure the whole if else block. I suppose I can add a nolint.

Reply:
OK nbd